Headers/DebugServer2/Host/File.h

```diff
@@ -45,6 +45,9 @@ class File {
   }
 
 public:
+  virtual ErrorCode pread(std::string &buf, uint64_t &count, uint64_t offset);
```

Comment:
If this is going to be a virtual method, then your destructor should be virtual as well.